src/qilib/utils/serialization.py

```diff
@@ -20,18 +20,68 @@
 import base64
 from functools import partial
 from json import JSONDecoder, JSONEncoder
-from typing import Any, Callable, Dict, Tuple, Optional
+from typing import Any, Callable, Dict, Tuple, Optional, Union
 from dataclasses_json.api import DataClassJsonMixin
-
 import numpy as np
 
-from qilib.data_set.mongo_data_set_io import MongoDataSetIO
+from qilib.utils.type_aliases import EncodedNumpyArray, numpy_ndarray_type, NumpyKeys
+
 
 # A callable type for transforming a given argument with a type to another type
 TransformFunctionResult = Any
 TransformFunction = Callable[[Any], TransformFunctionResult]
 
 
+class NumpyArrayEncDec:
+    """ Helper class for the numpy array decoder and encoder """
```

Comment:
uber nit pick: it looks like this class _is_ the actually encoder/decoder. Should we describe it as just "The numpy array decoder and encoder" or "Class to decode and encode numpy arrays"?

Reply:
Agree, I'll change